client/extensions/woocommerce/state/wc-api/products/reducer.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_PRODUCT_UPDATED,
+} from '../../action-types';
+
+export default {
+	[ WOOCOMMERCE_PRODUCT_UPDATED ]: productUpdated,
+};
+
+function productUpdated( siteData, action ) {
+	const { data } = action.payload;
+
+	return updateCachedProduct( siteData, data );
+}
+
+function updateCachedProduct( siteData, product ) {
+	const products = siteData.products || [];
+
+	let found = false;
+	const newProducts = products.map( ( p ) => {
```

Comment:
I've seen it elsewhere. And I'm already using `product` as the incoming argument. Is this a problem?